In [1]:
import pandas as pd

In [2]:
pd.read_csv('https://projects.publicsource.org/2018_statesalaries/state_data.csv')

agency       last_name  first_name  middle_name  \
0               Aging    Laudenslager       Kathy            A   
1               Aging        McDaniel       Megan           C.   
2               Aging       Burhannan        Rita          Gay   
3               Aging          Rhodes       Janis            L   
4               Aging          Schell       Trudy            L   
5               Aging           Allen      Ronald            R   
6               Aging             Fox        Abby            J   
7               Aging           Lorah     Rebecca           D.   
8               Aging          Paulus        Rose            M   
9               Aging          Horner      Steven            B   
10              Aging  Beers-Franklin       Tamea         Lynn   
11              Aging         Bennett     Lindsey          Rae   
12              Aging         Cenname  Anna Marie            E   
13              Aging          Cherry      Robert         John   
14              Aging         Claroni       Rocco            L   
15              Aging            Cree       Susan       Elaine   
16              Aging          Deppen     Barbara            G   
17              Aging         Dunston        Troy            D   
18              Aging         Ehrhart        Dana            M   
19              Aging          Gerber        Beth         Anne   
20              Aging         Glidden    Michelle            S   
21              Aging       Hollywood       Trula          NaN   
22              Aging            Kell    Veronica        Lynne   
23              Aging            Kyle     Katrina           K.   
24              Aging           Lomax       Dowel            E   
25              Aging          Nguyen        Uyen           P.   
26              Aging         Rodgers      Steven            L   
27              Aging          Rogers      Roslyn       Sylvia   
28              Aging            Seby       Julie    Elizabeth   
29              Aging            Toro      Carmen            M   
...               ...             ...         ...          ...   
89889  Transportation        Peterson        Ross            E   
89890  Transportation         Polanis     Jeffrey            F   
89891  Transportation         Proctor     Jeffrey          Ray   
89892  Transportation             Pyo     Anthony         John   
89893  Transportation           Rowan      Joseph            P   
89894  Transportation        Schwartz       David            A   
89895  Transportation          Seeley        John            D   
89896  Transportation          Sienko      Daniel            J   
89897  Transportation       Slaughter        Troy          NaN   
89898  Transportation           Smith    Clifford            W   
89899  Transportation          Sonney        Eric       August   
89900  Transportation        St Clair       Scott            P   
89901  Transportation           Tokay      Ronald            J   
89902  Transportation           Tomko     Michael            K   
89903  Transportation        Tridente     Corrado      Leonard   
89904  Transportation        Vranicar     Michael            D   
89905  Transportation          Wagner     Patrick            J   
89906  Transportation          Weikel       Ricky           R.   
89907  Transportation          Whalen      Joseph            H   
89908  Transportation        Williams      Joseph            T   
89909  Transportation         Withers       Jacob      Richard   
89910  Transportation           Womer       Ethan       Daniel   
89911  Transportation        Woodbury        Noah  Clifton Toy   
89912  Transportation            Wyne        Gary            I   
89913  Transportation           Zoria      Steven            A   
89914  Transportation        Bickford      Robert            E   
89915  Transportation       Eckenroth     Brandon      Michael   
89916  Transportation           Clark      Dennis           L.   
89917  Transportation       Kishbaugh      Donald      

In [3]:
statesal = pd.read_csv('https://projects.publicsource.org/2018_statesalaries/state_data.csv')

In [4]:
statesal.head()

agency     last_name first_name middle_name     job_name original_hire_dt  \
0  Aging  Laudenslager      Kathy           A       Acct 3          5/28/97   
1  Aging      McDaniel      Megan          C.  Admv Asst 1           3/7/05   
2  Aging     Burhannan       Rita         Gay   Admv Ofr 1          6/24/71   
3  Aging        Rhodes      Janis           L   Admv Ofr 1           8/2/93   
4  Aging        Schell      Trudy           L   Admv Ofr 1          9/29/75   

  incumbent_entered_pos separation_dt  hourly  base_salary  \
0               7/26/14           NaN     NaN      63785.0   
1               7/12/14           NaN     NaN      41956.0   
2              10/24/15           NaN     NaN      43775.0   
3               5/18/13           NaN     NaN      54690.0   
4               7/21/07           NaN     NaN      45829.0   

   actual_salary_2017  overtime_2017  leave_payout_2017  
0            62893.52            NaN                NaN  
1            41370.77            NaN                NaN  
2            43162.50            NaN                NaN  
3            53923.88            NaN                NaN  
4            45188.26            NaN                NaN

In [5]:
statesal.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 89919 entries, 0 to 89918
Data columns (total 13 columns):
agency                   89919 non-null object
last_name                89919 non-null object
first_name               77288 non-null object
middle_name              68077 non-null object
job_name                 89919 non-null object
original_hire_dt         89919 non-null object
incumbent_entered_pos    79380 non-null object
separation_dt            10539 non-null object
hourly                   12132 non-null float64
base_salary              77785 non-null float64
actual_salary_2017       88504 non-null float64
overtime_2017            51853 non-null float64
leave_payout_2017        7851 non-null float64
dtypes: float64(5), object(8)
memory usage: 8.9+ MB


In [6]:
statesal.actual_salary_2017.reset_index()

index  actual_salary_2017
0          0            62893.52
1          1            41370.77
2          2            43162.50
3          3            53923.88
4          4            45188.26
5          5            71804.63
6          6            64005.31
7          7            57643.89
8          8            93522.39
9          9           104420.25
10        10            60215.63
11        11            53923.88
12        12            22392.00
13        13            55063.52
14        14            28356.45
15        15            68678.27
16        16            56378.64
17        17            70227.38
18        18            54616.05
19        19            55063.51
20        20            59598.39
21        21            55063.51
22        22            49735.50
23        23            57643.89
24        24            53813.25
25        25            57643.89
26        26            55063.52
27        27            81894.38
28        28            56378.64
29        29            75009.01
...      ...                 ...
89889  89889            34542.38
89890  89890            35095.09
89891  89891            34552.07
89892  89892            34282.51
89893  89893            34246.53
89894  89894            43207.26
89895  89895            40978.15
89896  89896            40909.38
89897  89897             7555.51
89898  89898            47221.50
89899  89899            36220.27
89900  89900            39473.02
89901  89901            42984.93
89902  89902            14736.00
89903  89903            33060.33
89904  89904            40504.15
89905  89905            33033.80
89906  89906            37338.75
89907  89907            38979.81
89908  89908            41380.29
89909  89909             5641.49
89910  89910            32632.68
89911  89911            32373.28
89912  89912            40504.15
89913  89913            40557.48
89914  89914            33735.76
89915  89915            34042.50
89916  89916            38082.13
89917  89917            51660.01
89918  89918            42291.88

[89919 rows x 2 columns]

**Note: The above command would not work with column head 2017_actual_salary because it seems pandas does not like numbers to start column names. Tried moving date to end of column name, it worked.**

In [7]:
statesal.overtime_2017.sum()

233341250.74000004

**Note: The above command was not working, so I figured I had to change the data types of the columns. This video was helpful > https://www.youtube.com/watch?v=V0AWyzVMf54. But that still didn't work :( See below note for what I had to do because I reran the cells and it will all look like it magically worked the first time! ... Sadly, it did not, when I originally put the data in they were all dtype "object" which pretty much means "string" or "text", but we needed the columns with numerical data with decimals to be "floats".**

In [8]:
statesal.dtypes

agency                    object
last_name                 object
first_name                object
middle_name               object
job_name                  object
original_hire_dt          object
incumbent_entered_pos     object
separation_dt             object
hourly                   float64
base_salary              float64
actual_salary_2017       float64
overtime_2017            float64
leave_payout_2017        float64
dtype: object

**Note: Still not working, so in Excel, I converted columns I-M to "Number" and remove dollar signs and commas and finally got it working!**

In [9]:
actual_salary = statesal.actual_salary_2017

**Note: Above creates a variable which stores column actual_salary_2017 in it so we can run analysis on it instead of always have to type out statesal.actual_salary_2017, but we certainly could do that.**

In [10]:
actual_salary.median()

44640.509999999995

**Note: Above calculates overall employee median salary.**

In [11]:
actual_salary.sum()

4097021128.1000004

**Note: Above calculates total amount spent by state of Pennsylvania on empoloyee salaries.**

In [12]:
statesal.sort_values("actual_salary_2017", ascending=False).head()

agency last_name first_name middle_name  \
65060  Pub School Employees' Ret Sys  Grossman      James           H   
65080  Pub School Employees' Ret Sys   Spiller    Charles           J   
65079  Pub School Employees' Ret Sys     Bauer     Thomas          A.   
68277       State Employees' Ret Sys     Lewis    Wilbert       Bryan   
65146  Pub School Employees' Ret Sys    Devine     Robert           J   

                       job_name original_hire_dt incumbent_entered_pos  \
65060       Chf Invmt Ofr Psers          3/17/97               1/13/14   
65080   Dep Chf Invmt Ofr Psers          5/16/94               5/16/94   
65079   Dep Chf Invmt Ofr Psers          4/19/16               4/19/16   
68277        Chf Invmt Ofr Sers          7/11/16               7/11/16   
65146  Mgng Dir Invmt Off PSERS          6/10/91               9/29/14   

      separation_dt  hourly  base_salary  actual_salary_2017  overtime_2017  \
65060           NaN     NaN     407826.0           387184.88            NaN   
65080           NaN     NaN     335728.0           318274.50            NaN   
65079           NaN     NaN     335728.0           317697.38            NaN   
68277           NaN     NaN     286534.0           285655.50            NaN   
65146           NaN     NaN     271062.0           257344.13            NaN   

       leave_payout_2017  
65060                NaN  
65080                NaN  
65079                NaN  
68277                NaN  
65146                NaN

In [13]:
import pandas as pd

In [14]:
statesal.sort_values("actual_salary_2017", ascending=False)

agency          last_name   first_name  \
65060  Pub School Employees' Ret Sys           Grossman        James   
65080  Pub School Employees' Ret Sys            Spiller      Charles   
65079  Pub School Employees' Ret Sys              Bauer       Thomas   
68277       State Employees' Ret Sys              Lewis      Wilbert   
65146  Pub School Employees' Ret Sys             Devine       Robert   
65147  Pub School Employees' Ret Sys               Kemp         John   
65306  Pub School Employees' Ret Sys                 Oh        Susan   
65304  Pub School Employees' Ret Sys            Foreman       Darren   
65058  Pub School Employees' Ret Sys               Carl        Brian   
65303  Pub School Employees' Ret Sys               Derr       Steven   
65301  Pub School Employees' Ret Sys              Baker  Christopher   
47127                 Human Services             Behjat       Shahla   
65310  Pub School Employees' Ret Sys            Stalter      William   
65305  Pub School Employees' Ret Sys             Little       Robert   
65087  Pub School Employees' Ret Sys              Grell         Glen   
68298       State Employees' Ret Sys              Nolan        James   
68335       State Employees' Ret Sys              Felix        David   
65309  Pub School Employees' Ret Sys              Sheva       Joseph   
15711                    Corrections             Cooper          NaN   
65302  Pub School Employees' Ret Sys             Burton      Jeffrey   
65308  Pub School Employees' Ret Sys            Shaffer         Todd   
68304       State Employees' Ret Sys             Durbin        David   
5116                     Corrections               Noel         Paul   
31885              Governor'S Office               Wolf       Thomas   
68333       State Employees' Ret Sys              Meyer      Jeffrey   
47130                 Human Services               Shah        Hansa   
45801                 Human Services            Cherpes      Gregory   
68336       State Employees' Ret Sys          Greenberg      Phillip   
33926                 Human Services           Gopalani         Aziz   
47131                 Human Services            Wildner         Ralf   
...                              ...                ...          ...   
86805                 Transportation             Thiede      Patrice   
86806                 Transportation              Thoma      Kenneth   
86814                 Transportation             Thomas     Michelle   
86827                 Transportation             Thumma        Shane   
86836                 Transportation               Todd       Ashley   
86856                 Transportation             Troyer        Darin   
86936                 Transportation             Wanser       Robert   
86966                 Transportation       Webb-Lippert   Kristopher   
87034                 Transportation              Wills        Terry   
87062                 Transportation              Wolff      Jeffrey   
87063                 Transportation           Wolfgang        Jason   
87118                 Transportation           Zahniser       Gordon   
87120                 Transportation          Zarkevics        Daryl   
87419                 Transportation  Carrington-Setzer        Jason   
87774                 Transportation           Freiters        Barry   
87951                 Transportation         Hattersley         Paul   
87958                 Transportation            Hawkins       Lester   
88085                 Transportation           Ingerick         Tony   
88116                 Transportation              Jones        David   
88221                 Transportation           Kokoszka        Allan   
88227                 Transportation           Kozlosky       Andrew   
88369                 Transportation            Malizia        Floyd   
88411                 Transportation            McCarty       Ronald   
88446                 Transportation           Mckinley         Dale  

**Note: Above answers quesion who is highest paid state employee"**

In [15]:
statesal.groupby("agency").actual_salary_2017.sum().reset_index().sort_values("actual_salary_2017", ascending=False).astype(str)

agency  actual_salary_2017
6                      Corrections   904125528.1499993
20                  Human Services   785568557.2899544
34                  Transportation   566175907.7199867
33                    State Police   467153889.2099977
22                Labor & Industry  221732465.86999726
24            Liquor Control Board   161805897.5999996
12            Environmntl Protectn   150866342.5199989
7             Dept Miltry/Vets Aff  104756516.59999971
30                         Revenue  100555142.89999996
13               Executive Offices   96304194.61000004
5             Cnsvn & Natrl Resrcs   89700038.30999969
28      Probation And Parole Board   73117959.47000006
18                          Health    56188254.5600004
16                General Services   39964335.51999999
1                      Agriculture    37320019.0600001
15                 Game Commission   34822849.16000002
9                        Education  30433738.999999985
31                           State  26944304.530000016
29   Pub School Employees' Ret Sys   21345666.74000001
4              Cmunity & Econ Dvpt  20776928.069999985
14        Fish And Boat Commission         20105844.14
21                       Insurance  14347838.129999999
2           Banking and Securities  13069454.260000015
19  Historical & Museum Commission  11404598.479999997
10     Emergency Management Agency  10768091.259999998
32        State Employees' Ret Sys  10668052.290000001
3         Civil Service Commission   5726775.559999996
0                            Aging   5700945.559999992
17               Governor'S Office          4551938.02
8        Drug and Alcohol Programs  4051355.7899999996
27                        Pennvest          2158865.29
26      Municipal Retirement Board  1554522.3699999996
25            Milk Marketing Board          1275789.24
11     Environmental Hearing Board          1150835.49
23    Lieutenant Governor'S Office   827685.3300000001

**Note: Above cell answers question of top agencies paying out salaries**

**Hi, this is Dale jumping into this analysis. What you'll see below is me trying to figure out how to write some basic Excel funcions in Python.**

In [16]:
statesal.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 89919 entries, 0 to 89918
Data columns (total 13 columns):
agency                   89919 non-null object
last_name                89919 non-null object
first_name               77288 non-null object
middle_name              68077 non-null object
job_name                 89919 non-null object
original_hire_dt         89919 non-null object
incumbent_entered_pos    79380 non-null object
separation_dt            10539 non-null object
hourly                   12132 non-null float64
base_salary              77785 non-null float64
actual_salary_2017       88504 non-null float64
overtime_2017            51853 non-null float64
leave_payout_2017        7851 non-null float64
dtypes: float64(5), object(8)
memory usage: 8.9+ MB


In [17]:
ot = statesal.overtime_2017

In [18]:
ot.sum()

233341250.74000004

**In the above cells I was getting my bearings with the data we're working with, then I defined "ot" as the variable we'll use for analyses of overtime pay and I found that total overtime pay added up to $233,341,250.74.**

In [19]:
statesal.groupby("agency").overtime_2017.sum().reset_index().sort_values("overtime_2017", ascending=False).astype(str)

agency       overtime_2017
6                      Corrections   91576468.60000035
34                  Transportation   52335062.31999996
20                  Human Services   34669246.76000013
33                    State Police  30580563.689999938
24            Liquor Control Board   5328969.029999994
7             Dept Miltry/Vets Aff   4262474.690000001
22                Labor & Industry   3806418.349999994
5             Cnsvn & Natrl Resrcs   2419581.859999993
28      Probation And Parole Board  1663564.4799999986
15                 Game Commission   1478888.600000002
1                      Agriculture   797576.9799999995
14        Fish And Boat Commission   783714.4600000001
12            Environmntl Protectn   718697.6299999997
16                General Services   671351.7500000001
18                          Health  435647.94000000006
10     Emergency Management Agency  353290.61000000016
30                         Revenue           298728.07
29   Pub School Employees' Ret Sys           275618.21
21                       Insurance            268002.3
31                           State   251543.8099999999
13               Executive Offices  119140.73000000004
19  Historical & Museum Commission   82203.65000000001
8        Drug and Alcohol Programs   59703.39999999999
9                        Education   56590.08000000001
32        State Employees' Ret Sys            18594.63
27                        Pennvest  16545.809999999998
3         Civil Service Commission             4997.54
26      Municipal Retirement Board   4621.540000000001
4              Cmunity & Econ Dvpt             1770.42
2           Banking and Securities             1381.47
23    Lieutenant Governor'S Office              195.32
0                            Aging               96.01
25            Milk Marketing Board                 0.0
11     Environmental Hearing Board                 0.0
17               Governor'S Office                 0.0

In [20]:
statesal.sort_values("actual_salary_2017", ascending=False)

agency          last_name   first_name  \
65060  Pub School Employees' Ret Sys           Grossman        James   
65080  Pub School Employees' Ret Sys            Spiller      Charles   
65079  Pub School Employees' Ret Sys              Bauer       Thomas   
68277       State Employees' Ret Sys              Lewis      Wilbert   
65146  Pub School Employees' Ret Sys             Devine       Robert   
65147  Pub School Employees' Ret Sys               Kemp         John   
65306  Pub School Employees' Ret Sys                 Oh        Susan   
65304  Pub School Employees' Ret Sys            Foreman       Darren   
65058  Pub School Employees' Ret Sys               Carl        Brian   
65303  Pub School Employees' Ret Sys               Derr       Steven   
65301  Pub School Employees' Ret Sys              Baker  Christopher   
47127                 Human Services             Behjat       Shahla   
65310  Pub School Employees' Ret Sys            Stalter      William   
65305  Pub School Employees' Ret Sys             Little       Robert   
65087  Pub School Employees' Ret Sys              Grell         Glen   
68298       State Employees' Ret Sys              Nolan        James   
68335       State Employees' Ret Sys              Felix        David   
65309  Pub School Employees' Ret Sys              Sheva       Joseph   
15711                    Corrections             Cooper          NaN   
65302  Pub School Employees' Ret Sys             Burton      Jeffrey   
65308  Pub School Employees' Ret Sys            Shaffer         Todd   
68304       State Employees' Ret Sys             Durbin        David   
5116                     Corrections               Noel         Paul   
31885              Governor'S Office               Wolf       Thomas   
68333       State Employees' Ret Sys              Meyer      Jeffrey   
47130                 Human Services               Shah        Hansa   
45801                 Human Services            Cherpes      Gregory   
68336       State Employees' Ret Sys          Greenberg      Phillip   
33926                 Human Services           Gopalani         Aziz   
47131                 Human Services            Wildner         Ralf   
...                              ...                ...          ...   
86805                 Transportation             Thiede      Patrice   
86806                 Transportation              Thoma      Kenneth   
86814                 Transportation             Thomas     Michelle   
86827                 Transportation             Thumma        Shane   
86836                 Transportation               Todd       Ashley   
86856                 Transportation             Troyer        Darin   
86936                 Transportation             Wanser       Robert   
86966                 Transportation       Webb-Lippert   Kristopher   
87034                 Transportation              Wills        Terry   
87062                 Transportation              Wolff      Jeffrey   
87063                 Transportation           Wolfgang        Jason   
87118                 Transportation           Zahniser       Gordon   
87120                 Transportation          Zarkevics        Daryl   
87419                 Transportation  Carrington-Setzer        Jason   
87774                 Transportation           Freiters        Barry   
87951                 Transportation         Hattersley         Paul   
87958                 Transportation            Hawkins       Lester   
88085                 Transportation           Ingerick         Tony   
88116                 Transportation              Jones        David   
88221                 Transportation           Kokoszka        Allan   
88227                 Transportation           Kozlosky       Andrew   
88369                 Transportation            Malizia        Floyd   
88411                 Transportation            McCarty       Ronald   
88446                 Transportation           Mckinley         Dale  

In [21]:
pd.set_option('display.max_rows', 1000)

In [22]:
statesal.sort_values("actual_salary_2017", ascending=False)

agency          last_name    first_name  \
65060  Pub School Employees' Ret Sys           Grossman         James   
65080  Pub School Employees' Ret Sys            Spiller       Charles   
65079  Pub School Employees' Ret Sys              Bauer        Thomas   
68277       State Employees' Ret Sys              Lewis       Wilbert   
65146  Pub School Employees' Ret Sys             Devine        Robert   
65147  Pub School Employees' Ret Sys               Kemp          John   
65306  Pub School Employees' Ret Sys                 Oh         Susan   
65304  Pub School Employees' Ret Sys            Foreman        Darren   
65058  Pub School Employees' Ret Sys               Carl         Brian   
65303  Pub School Employees' Ret Sys               Derr        Steven   
65301  Pub School Employees' Ret Sys              Baker   Christopher   
47127                 Human Services             Behjat        Shahla   
65310  Pub School Employees' Ret Sys            Stalter       William   
65305  Pub School Employees' Ret Sys             Little        Robert   
65087  Pub School Employees' Ret Sys              Grell          Glen   
68298       State Employees' Ret Sys              Nolan         James   
68335       State Employees' Ret Sys              Felix         David   
65309  Pub School Employees' Ret Sys              Sheva        Joseph   
15711                    Corrections             Cooper           NaN   
65302  Pub School Employees' Ret Sys             Burton       Jeffrey   
65308  Pub School Employees' Ret Sys            Shaffer          Todd   
68304       State Employees' Ret Sys             Durbin         David   
5116                     Corrections               Noel          Paul   
31885              Governor'S Office               Wolf        Thomas   
68333       State Employees' Ret Sys              Meyer       Jeffrey   
47130                 Human Services               Shah         Hansa   
45801                 Human Services            Cherpes       Gregory   
68336       State Employees' Ret Sys          Greenberg       Phillip   
33926                 Human Services           Gopalani          Aziz   
47131                 Human Services            Wildner          Ralf   
47128                 Human Services                Liu         Danny   
5119                     Corrections              Reddy      Palukuri   
12684                    Corrections              Moody           NaN   
47129                 Human Services               Ravi      Sasikala   
65153  Pub School Employees' Ret Sys             Jacobs          Luke   
21470           Dept Miltry/Vets Aff            Jackson        Darryl   
33950                 Human Services          Martyniuk          John   
65150  Pub School Employees' Ret Sys         Del Gaudio         James   
65081  Pub School Employees' Ret Sys            Sanchez       Terrill   
50277                 Human Services             Makkar         Louka   
13633                    Corrections            Roberts           NaN   
45988                 Human Services               Murr       Phyllis   
50276                 Human Services         Lightbourn         Susan   
50280                 Human Services              Yager         Maria   
50279                 Human Services               Shah        Asmita   
65155  Pub School Employees' Ret Sys              Tyler       Michael   
50359                 Human Services           Fakiyesi           Olu   
65151  Pub School Employees' Ret Sys             Engler        Lenann   
65152  Pub School Employees' Ret Sys              Etter        Andrea   
65154  Pub School Employees' Ret Sys            Sellers      Margaret   
29654              Executive Offices           Albright      Randolph   
33928                 Human Services            Richard        Andrea   
56416   Lieutenant Governor'S Office              Stack       Michael   
50364                 Human Services             Prasad   Mahalingiah   
50365          

In [23]:
statesal.actual_salary_2017 > 100,000

(0         True
 1         True
 2         True
 3         True
 4         True
 5         True
 6         True
 7         True
 8         True
 9         True
 10        True
 11        True
 12        True
 13        True
 14        True
 15        True
 16        True
 17        True
 18        True
 19        True
 20        True
 21        True
 22        True
 23        True
 24        True
 25        True
 26        True
 27        True
 28        True
 29        True
 30        True
 31        True
 32        True
 33        True
 34        True
 35        True
 36        True
 37        True
 38        True
 39        True
 40        True
 41        True
 42        True
 43        True
 44        True
 45        True
 46        True
 47        True
 48        True
 49        True
 50        True
 51        True
 52        True
 53        True
 54        True
 55        True
 56        True
 57        True
 58        True
 59        True
 60        True
 61        True
 62     

**Above is me trying to figure out how to calculate salary ranges and failing, but I figure it out below! :(**

In [24]:
actual_salary.count()

88504

**Above counts number of people who received a salary in 2017.**

In [25]:
actual_salary[actual_salary > 200000].count() 

21

In [26]:
actual_salary[actual_salary > 100000][actual_salary < 200000].count() 

2591

In [27]:
actual_salary[actual_salary > 80000][actual_salary < 100000].count()

5465

In [28]:
actual_salary[actual_salary > 60000][actual_salary < 80000].count()

16171

In [29]:
actual_salary[actual_salary > 60000][actual_salary < 80000].count()

16171

In [30]:
actual_salary[actual_salary > 40000][actual_salary < 60000].count()

29225

In [31]:
actual_salary[actual_salary > 20000][actual_salary < 40000].count()

20352

In [32]:
actual_salary[actual_salary < 20000].count() 

14678

**Above eight cells calculate the number of employees within each salary range.**